### PR TITLE
Fixes #1038 - Sign the Core tools binaries

### LINKED ISSUE #1038: Core Tools V2 binaries must be signed
V2 binaries are not currently signed. We need to match the behavior we had in V1 and ensure those are signed with the same process.

### PR DESCRIPTION
**Pull request by ankitkumarr**
The signing job is only for master branch right now.

~~@ahmelsayed, we also need to add the connection string in Appveyor.~~ Thanks Ahmed!

### CODE REVIEW

File: build/Settings.cs [hunk 1]
```diff
@@ -56,5 +57,76 @@ private static string config(string @default = null, [CallerMemberName] string k
...
+
+            public static readonly string[] authentiCodeBinaries = new[] {
+                "DurableTask.AzureStorage.Internal.dll",
+                "DurableTask.Core.Internal.dll",
+                "func.deps.json",
+                "func.dll",
+                "func.dll.config",
+                "func.exe",
+                "func.pdb",
+                "func.runtimeconfig.json",
```
Comment (ahmelsayed) [inline build/Settings.cs]: non-executables like `.json`, `.config` etc don't have to be signed. executable scripts like `.js` and `.py` do need to be signed.

File: build/Settings.cs [hunk 2]
```diff
@@ -56,5 +57,76 @@ private static string config(string @default = null, [CallerMemberName] string k
...
+                "Microsoft.Azure.WebJobs.Extensions.Http.dll",
+                "Microsoft.Azure.WebJobs.Host.dll",
+                "Microsoft.Azure.WebJobs.Host.Storage.dll",
+                "Microsoft.Azure.WebJobs.Logging.ApplicationInsights.dll",
+                "Microsoft.Azure.WebJobs.Logging.dll",
+                "Microsoft.Azure.WebJobs.Script.dll",
+                "Microsoft.Azure.WebJobs.Script.Grpc.dll",
+                "Microsoft.Azure.WebJobs.Script.WebHost.dll",
+                "Microsoft.Azure.WebSites.DataProtection.dll",
+                Path.Combine("templates", "itemTemplates.2.0.10328.nupkg"),
```
Comment (ahmelsayed) [inline build/Settings.cs]: @mhoeger  is the signing process able to sign nupkgs in the zip file directly?

Comment (mhoeger) [reply build/Settings.cs]: Depends on the signing... for host, we strong name sign all of the .dll's in the nupkg and then nuget sign the entire package. I don't know how Authenticode signing works with nupkgs. @ahmelsayed, i think you or fabio would know more on that! 

So to answer your question: not sure if nupkgs are able to be authenticode signed, but we do have a process for nuget signing and/or strong name signing the .dll's

File: build/Settings.cs (hunk 1 above)
Comment (ankitkumarr) [reply build/Settings.cs]: The `.py` files in distlib were failing with the third party signing scripts, and it sounded like we neede d the `.dll`s to be signed asap, so for now, I ended us ignoring `.py` files.

File: build/Settings.cs (hunk 2 above)
Comment (ankitkumarr) [reply build/Settings.cs]: @ahmelsayed, I made these lists of the unsigned things and then filtered out things that we weren't able to sign from the scripts. Just so we have the lists properly and we are taking into account all the files.

File: build/Settings.cs (hunk 1 above)
Comment (ahmelsayed) [reply build/Settings.cs]: Just thought it was a bit misleading for .json and .config files to be in the list, but not a big deal.

File: build/Settings.cs (hunk 2 above)
Comment (ahmelsayed) [reply build/Settings.cs]: oh I see, sorry I missed the filtering below.

File: build/Settings.cs (hunk 1 above)
Comment (ankitkumarr) [reply build/Settings.cs]: I think you are right, I will remove those files from the list

Comment (ankitkumarr) [reply build/Settings.cs]: I updated the PR and removed those files.

File: build/Settings.cs (hunk 2 above)
Comment (fabiocav) [reply build/Settings.cs]: For clarity, only the contents need to be authenticode signed. The nupkg should be signed like the others.

**Pull request merged**